Source/render.cpp

```diff
@@ -181,7 +181,7 @@ void RenderTile(BYTE *pBuff)
 		if (level_cel_block & 0x8000) {
 			level_cel_block = SpeedFrameTbl[level_cel_block & 0xFFF][0] + (level_cel_block & 0xF000);
 		}
-		src = &pDungeonCels[pFrameTable[level_cel_block & 0xFFF]];
+		src = &pDungeonCels[SDL_SwapLE32(pFrameTable[level_cel_block & 0xFFF])];
 		tile = (level_cel_block & 0x7000) >> 12;
 		tbl = NULL;
 	} else if (level_cel_block & 0x8000) {

```

Comment:
This:
`src = &pSpeedCels[SpeedFrameTbl[level_cel_block & 0xFFF][light_table_index]];`
Need to be this:
`src = &pSpeedCels[SDL_SwapLE32(SpeedFrameTbl[level_cel_block & 0xFFF][light_table_index])];`

Reply:
Is USE_SPEEDCELS set by default ?

Reply:
Yes, it's in the top of the file, with out it town works but dungion crashes

Reply:
On my build I could go to dungeon 4 , where it crashed. 
Dungeon 1,2 and 3 worked.

Reply:
For ex. in town.cpp  Aug 13 and Aug 16
https://github.com/AmigaPorts/devilutionX/commits/master/Source/town.cpp

Reply:
How does it work with the suggested changes? How did you get to the dungeons? For me the crash was during the load screen, no debug just normal game play.

Reply:
I could go to dungeons in linux build too.

Reply:
Turns out render.cpp was correct, only `makespeedcels` needed to be changed.

Reply:
I added that define to the renderer because speed cels are uncessary. However the function `MakeSpeedCels` has to be removed or else it screws up the table and crashes. I believe Brevik mentioned speed cels in an interview once, a way to increase render speed. But after benchmarking it doesn't help, it actually increases the loading screen by ~100k nanoseconds.

Reply:
`~100k nanoseconds.` - that's a fancy way of saying 0.1 sec :D

Reply:
Could also just be 100,000,000 picoseconds divided by 0.3 light years. Time and space are one...

Reply:
Ok, will try and disable both